src/documents/models.py

```diff
@@ -267,17 +277,90 @@ def __str__(self):
         return str(created)
 
     @property
-    def source_path(self):
-
-        file_name = "{:07}.{}".format(self.pk, self.file_type)
+    def source_filename(self):
+        if self.filename is None:
+            self.filename = self.source_filename_new()
+
+        return self.filename
+
+    def many_to_list(self, field):
+        mylist = []
+        for t in field.all():
+            mylist.append(t.name)
+        return mylist
+
+    def many_to_dictionary(self, field):
+        mydictionary = dict()
+        for t in field.all():
+            delimeter = t.name.find('_')
+
+            if delimeter is -1:
+                continue
+
+            key = t.name[:delimeter]
+            value = t.name[delimeter+1:]
+
+            mydictionary[key] = value
+
+        return mydictionary
```

Comment:
Indeed, this was not properly documented. Basically, we have two ways to use tags in filenames:
* When your tags have the format KEY-VALUE or KEY_VALUE, then tag[KEY] selects this particular tag
* Otherwise, you can still select tags by index by using tags[INDEX]